### ISSUE TITLE
Question related to the model training and usage of raw text

### ISSUE THREAD

**Issue by StefanusAgus** [kongds/LightXML#11]
Hello. Thank you very much for the impressive work and for sharing your work. I have one question related to the training process. In the paper, I read that LightXML can use raw text data to provide end-to-end prediction, similar to usual deep learning based approaches which use raw text.

However, when I ran the LightXML code, I found that TfIdf feature is used in the train.txt file. I wonder why this TfIdf is necessary? Would the TfIdf affect the performance? Can we swap the TfIdf with other vectorizer (e.g., bag of word feature, word piece, etc.).

Thank you very much for the help!

**Comment by kongds** [comment]
Hi
TfIdf is used to embedding each label. Then we use these label embedding to cluster. The only influence of Tfidf in our work is the label clusters.  You can replace it with other features to get sentence embedding like word embedding or sentence embedding model based on BERT.

**Comment by StefanusAgus** [comment]
Thank you very much for the clarification.

Would it be correct to say that the bert transformer is used to create the text representation that is used for the model training? Does the model consist of two layers of linear neural network?

From my understanding now, would it mean that if there are only small number of labels (i.e., less than 10k labels), LightXML is set to not do the label clustering and would not need the train.txt file. Is my understanding correct? Thank you!

**Comment by kongds** [comment]
Yes, BERT is used to create the text representation by concatenating the last five layers [CLS] tokens.

For linear neural network, we use the two linear neural network in large datasets, one for classification group, another is to project the text representation into low dimension. For small datasets, we use one linear as label classification.

For the Eurlex-4k, AmazonCat-13k and Wiki10-31K, we don't use the label clustering same with AttentionXML. So it don't need the train.txt file.
